blocks/product-support/product-support.js

```diff
@@ -0,0 +1,152 @@
+import { createDomStructure, translate } from '../../scripts/lib-franklin.js';
+
+function getInfo() {
+  const url = new URL(window.location);
+  const idx = url.pathname.indexOf('/product-support/');
+  if (idx > 0) {
+    const slug = url.pathname.substring(url.pathname.indexOf('/product-support/') + '/product-support/'.length);
+    if (slug) {
+      return {
+        productCode: slug,
+        productSupport: url.pathname.substring(0, url.pathname.indexOf(`${slug}`)),
+        language: url.pathname.substring(1, url.pathname.indexOf('/', 1)),
+      };
+    }
+  }
+  return { productSupport: url.pathname.substring(0, url.pathname.length - 1) };
+}
+
+function getProduct(json, productCode, language) {
+  const product = json.Product.data
+    .find((entry) => entry.ProductCodes.split('|')
+      .some((code) => code === productCode)
+        && entry.Languages.split('|')
+          .some((productLanguage) => productLanguage.toUpperCase() === language.toUpperCase()));
+
+  if (product) {
+    const translation = json.ProductTranslation.data
+      .find((entry) => entry.ProductRef === product.ProductCodes
+        && entry.Language === language);
+    if (translation) {
+      if (translation.Name) {
+        product.Name = translation.Name;
+      }
+      if (translation.Image) {
+        product.Image = translation.Image;
+      }
+    }
+  }
+
+  return product;
+}
+
+function getTypes(json, product, language) {
+  return json.ProductAsset.data
+    .filter((asset) => asset.ProductRef === product.ProductCodes && asset.Languages
+      .split('|').some((assetLanguage) => assetLanguage.toUpperCase() === language.toUpperCase()))
+    .map((asset) => asset.Type)
+    .filter((value, index, array) => array.indexOf(value) === index);
+}
+
+function getAssets(json, product, language, type, allType) {
+  return json.ProductAsset.data
+    .filter((asset) => asset.ProductRef === product.ProductCodes && asset.Languages
+      .split('|').some((assetLanguage) => assetLanguage.toUpperCase() === language.toUpperCase())
+      && (type === allType || asset.Type === type));
+}
+
+export default async function decorate(block) {
+  const resp = await fetch('/products.json?limit=10000');
+  if (!resp.ok) {
+    throw new Error(`${resp.status}: ${resp.statusText}`);
+  }
+
+  const { productCode, productSupport, language } = getInfo();
+
+  const json = await resp.json();
+
+  const product = getProduct(json, productCode, language);
+
+  if (!product) {
+    window.location.replace(productSupport);
+    return;
+  }
+
+  const heading = await translate('productSupportHeading', 'Product and Technical documents');
```

Comment:
could be combined for readability (matter of taste):
```
  const [heading, allDocuments, empty] = await Promise.all([
    translate('productSupportHeading', 'Product and Technical documents'),
    translate('productSupportAllDocuments', 'All Product Documents'),
    translate('productSupportNoResult', 'No data was found'),
  ]);
```

Reply:
Done